soc/silabs/silabs_s2/soc_image_properties.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#if defined __has_include
+#if __has_include("app_version.h")
+#include "app_version.h"
+#define SOC_IMAGE_VERSION APPVERSION
+#endif
+#endif
+
+#if !defined(SOC_IMAGE_VERSION)
+#include <zephyr/version.h>
+#define SOC_IMAGE_VERSION KERNELVERSION
+#endif
+
+#if !defined(SOC_IMAGE_TYPE)
```

Comment:
How the user is supposed to overload `SOC_IMAGE_TYPE`, `SOC_IMAGE_CAPABILITIES`, etc...?

Reply:
This doesn't implement a nice way of doing that, since I don't see a clean way of setting the 16-byte ID from Kconfig with its limited data types. So I just left an ifndef wrapper to allow those who really want to do it to inject a compile definition to override it. Not the cleanest interface, alternate implementation suggestions welcome.

Reply:
I assume most of the user will copy-paste it in their application. So, I assume we could just have the struct with a comment:
```
	/* Contains arbitrary values. The user may want to change these bits */
	.info = {
		.type = IMAGE_PROPERTY_TYPE,
		.version = IMAGE_PROPERTY_VERSION,
		.capabilities = 0,
		.id = { 0 },
	},
```